dotNETStandard/HealthVault.Mobile/cs/Xamarin.Core/ViewModels/MenuViewModel.cs

```diff
@@ -125,10 +126,31 @@ public class MenuViewModel : ViewModel
             IThingClient thingClient = this.connection.CreateThingClient();
             HealthRecordInfo record = person.SelectedRecord;
             IReadOnlyCollection<Medication> items = await thingClient.GetThingsAsync<Medication>(record.Id);
+            IVocabularyClient vocabClient = this.connection.CreateVocabularyClient();
+            var ingredientChoices = new List<VocabularyItem>();
+
+            Vocabulary ingredientVocabulary = null;
+            while (ingredientVocabulary == null || ingredientVocabulary.IsTruncated)
+            {
+                string lastCodeValue = null;
+                if (ingredientVocabulary != null)
+                {
+                    lastCodeValue = ingredientVocabulary.Values.Last().Value;
```

Comment:
Are you sure that ingredientVocabulary.Values will have at least one item? Maybe use LastOrDefault?